frontend/appflowy_tauri/src/appflowy_app/components/document/ImageBlock/useImageBlock.ts

```diff
@@ -0,0 +1,186 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { Align, BlockType, NestedBlock } from '$app/interfaces/document';
+import { useAppDispatch } from '$app/stores/store';
+import { useSubscribeDocument } from '$app/components/document/_shared/SubscribeDoc.hooks';
+import { updateNodeDataThunk } from '$app_reducers/document/async-actions';
+import { Log } from '$app/utils/log';
+import { getNode } from '$app/utils/document/node';
+import { readImage } from '$app/utils/document/image';
+
+export function useImageBlock(node: NestedBlock<BlockType.ImageBlock>) {
+  const { url, width, align, height } = node.data;
+  const dispatch = useAppDispatch();
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<boolean>(false);
+  const { controller } = useSubscribeDocument();
+  const [resizing, setResizing] = useState<boolean>(false);
+  const startResizePoint = useRef<{
+    left: boolean;
+    x: number;
+    y: number;
+  }>();
+  const startResizeWidth = useRef<number>(0);
+
+  const [src, setSrc] = useState<string>('');
+  const [displaySize, setDisplaySize] = useState<{
+    width: number;
+    height: number;
+  }>({
+    width: width || 0,
+    height: height || 0,
+  });
+
+  const onResizeStart = useCallback(
+    (e: React.MouseEvent<HTMLDivElement>, left: boolean) => {
+      e.preventDefault();
+      e.stopPropagation();
+      setResizing(true);
+      startResizeWidth.current = displaySize.width;
+      startResizePoint.current = {
+        x: e.clientX,
+        y: e.clientY,
+        left,
+      };
+    },
+    [displaySize.width]
+  );
+
+  const updateWidth = useCallback(
+    (width: number, height: number) => {
+      dispatch(
+        updateNodeDataThunk({
+          id: node.id,
+          data: {
+            width,
+            height,
+          },
+          controller,
+        })
+      );
+    },
+    [controller, dispatch, node.id]
+  );
+
+  useEffect(() => {
+    const currentSize: {
+      width?: number;
+      height?: number;
+    } = {};
+    const onResize = (e: MouseEvent) => {
+      const clientX = e.clientX;
+
+      if (!startResizePoint.current) return;
+      const { x, left } = startResizePoint.current;
+      const startWidth = startResizeWidth.current || 0;
+      const diff = (left ? x - clientX : clientX - x) / 1.5;
```

Comment:
what does the `1.5` mean here?

Should it be `2`? I suppose the `image width = image width - drag offset * 2`.